test/CMakeLists.txt

```diff
@@ -1,2 +1,5 @@
-add_test(NAME tuple_keydef.test.lua
-         COMMAND runtest.sh tuple_keydef.test.lua)
+add_test(
+    NAME tuple_keydef.test.lua
+    COMMAND runtest.sh tuple_keydef.test.lua
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
```

Comment:
What is the problem you fixed here?

Reply:
The tests used to fail in our usual workflow:

```bash
tarantoolctl rocks make
make test -C build.luarocks
```

Reply:
In fact, it is only partial fix of `make test` for out of source build: the module is found by the test only because of the upward traversal for `.rocks` directory. Just out of source build without luarocks still fails.

I'll fix it like so:

```diff
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 9fc4e1d..3e79a48 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,2 +1,6 @@
-add_test(NAME tuple_keydef.test.lua
-         COMMAND runtest.sh tuple_keydef.test.lua)
+add_test(
+    NAME tuple_keydef.test.lua
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/runtest.sh tuple_keydef.test.lua)
+
+set_tests_properties(tuple_keydef.test.lua PROPERTIES ENVIRONMENT
+    "LUA_CPATH=${CMAKE_CURRENT_BINARY_DIR}/../?${CMAKE_SHARED_LIBRARY_SUFFIX};;")
```

Reply:
Why `..` in `${CMAKE_CURRENT_BINARY_DIR}/../`?

Reply:
Current is `test/`, while we need the root.

We can use `${CMAKE_BINARY_DIR}` or `${PROJECT_BINARY_DIR}`, but it only matters in context of reusing this CMake file in another project. And, in this case, `${CMAKE_CURRENT_BINARY_DIR}/..` works for sure, disregarding whether we call add_subdirectory() for the project root or just test directory.

It does not matter much: unlikely anybody will use it this way. But `${CMAKE_CURRENT_BINARY_DIR}/..` works in an obvious way, at least.

Reply:
Moved to PR #12.

Reply:
(See also PR #14.)